lib/daru/vector.rb

```diff
@@ -912,6 +911,28 @@ def reindex! new_index
       self
     end
 
+    # Reorder the vector
+    # @param [Array] order the order to reorder the vector with
+    # @return reordered vector
+    # @example
+    #   dv = Daru::Vector.new [3, 2, 1], index: ['c', 'b', 'a']
+    #   dv.reorder! [2, 1, 0]
+    #   # => #<Daru::Vector(3)>
+    #   #   a   1
+    #   #   b   2
+    #   #   c   3
+    def reorder! order
```

Comment:
So reorder works with positional index and reindex with actual index? Am I right?